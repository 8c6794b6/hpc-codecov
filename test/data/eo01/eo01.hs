{-# LANGUAGE GeneralizedNewtypeDeriving #-}
module Main where

data R = R
  { field1 :: Bool
  , field2 :: Bool
  }

pm01 :: Maybe Int -> Int
pm01 (Just n) =
  n + 1
pm01 Nothing =
  12345

br01 :: Int -> Int -> Int
br01 a b
  | a < b = b
  | a == 0 = b
  | b < 0 = a + b
  | otherwise = 12345

newtype A f x = A {unA :: f x}
  deriving (Functor, Applicative, Monad)

newtype B f x = B {unB :: f x}

instance Functor f => Functor (B f) where
  fmap f (B fx) = B (fmap f fx)

br02 :: Int -> Int -> Int -> Int
br02 a b c
  | a < b = b
  | a == 0 = b
  | b < c = a + b
  | otherwise = 12345

main :: IO ()
main = do
  case R True False of
    R t f -> print (t && f)
  print $ pm01 Nothing
  print $ br01 2 1
  print $ br02 0 (-1) 0
