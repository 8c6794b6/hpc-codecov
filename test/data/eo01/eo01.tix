Tix [ TixModule "Main" 1101637578 105 [1,1,0,1,1,0,1,1,1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,1,0,1,1,0,1,1,0,1,1,0,1,1,0,0,0,1,0,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]
