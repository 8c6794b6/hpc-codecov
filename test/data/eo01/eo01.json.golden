{"coverage":{"test/data/eo01/eo01.hs":{"5":0,"6":0,"11":0,"13":1,"17":"1/2","18":"1/2","19":"1/2","20":1,"22":0,"25":0,"28":0,"32":"1/2","33":1,"34":0,"35":0,"38":1,"39":1,"40":1,"41":1,"42":1,"43":"1/2"}}}
